service/matching/tasklist/task_list_limiter.go

```diff
@@ -66,9 +68,10 @@ func (l *taskListLimiter) Limit() rate.Limit {
 func (l *taskListLimiter) ReportLimit(rps float64) {
 	now := l.timeSource.Now()
 	// Optimistically reject it without locking if it's >= current and within the TTL
+	// unless it is the overrideRPS
 	if now.Sub(l.lastUpdate.Load()) < l.ttl {
 		current := l.value.Load()
-		if rps > current {
+		if rps > current && (l.overrideRPS() <= 0 || rps != l.overrideRPS()) {
```

Comment:
Your change here is the most correct way to implement this, but I think for the sake of simplicity we don't need to add the override check here. The override applying to the incoming values is good enough.

If the override is smaller than the current RPS then it'll immediately take, and if it's larger then it'll take after the TTL.